### PR TITLE
Quantity check, empty ingredient check, clear tags feature, modify isSameRecipe

### LINKED ISSUE #214: [PE-D] Missing space before hyphen for quantity field in ingredient argument does not throw an error.
Command to replicate bug: 
addR n/salad i/lettuce-2 whole  c/40 img/images/healthy1.jpg instr/Cook pasta. Serve immediately. t/fast t/easy

Command with bug:
Add recipe command

If the user enters this command, the application does not provide any response or results to inform if the input is accepted or rejected. Even though users are supposed to put a whitespace before the hyphen, there should be an error reply to alert users. In this case, there was no reply or response from the application at all.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: jonasngs/ped#2

### LINKED ISSUE #215: [PE-D] Entering a comma as an ingredient name is allowed, even though the application specified otherwise.
Command to replicate bug:
addR n/salad i/, c/40 img/images/healthy1.jpg instr/Cook pasta. Serve immediately. t/fast t/easy

Command with bug:
* add recipe command
* edit recipe command

If the user enters this command, a new recipe without ingredients will be added. The ingredient section shows “ , “. However, the error handling of the application specifies that ingredients should only contain alphanumeric characters so I do not think that the above command should allow users to add a new recipe since a comma is not an alphanumeric character.

Screenshot showing the requirements of ingredient names:

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: jonasngs/ped#3

### LINKED ISSUE #217: [PE-D] Missing hyphen before quantity field in add ingredient to fridge command does not provide errors
Command to replicate bug: 
addF i/tomato-1 kg

Command with bug: Adding an ingredient: `addF` under fridge related command

If the user enters the above command by accident without leaving a space before the hyphen in the quantity field, the application does not display any response or results, even though the command format is incorrect. The application should at least display an error message to the users.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: jonasngs/ped#5

### LINKED ISSUE #229: [PE-D] Recommend when not enough ingredients
"Recommend" command recommends me recipes when I do not have enough ingredients.

Steps to reproduce:
1. Add ingredient with 0 quantity.
2. Add recipe with >0 quantity.
3. recommend

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: todoge/ped#10

### PR DESCRIPTION
**Pull request by oliviajohansen**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/recipe/AddRecipeCommand.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
             + "[" + PREFIX_TAG + "TAG]...\n"
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_NAME + "Chicken salad "
-            + PREFIX_INGREDIENT + "chicken - 100g, lettuce - a bit, tomato "
+            + PREFIX_INGREDIENT + "chicken - 100.50g, lettuce - a bit, tomato -3/4 cups"
```
Comment (caitlinjee) [inline src/main/java/seedu/address/logic/commands/recipe/AddRecipeCommand.java]: in the first quantity, there a space "-(space)100.5g", but in the 2nd one there isnt "-3/4 cups", maybe can standardize?

File: src/main/java/seedu/address/model/recipe/Recipe.java [hunk 2]
```diff
@@ -94,11 +94,21 @@ public boolean isSameRecipe(Recipe otherRecipe) {
                 && otherRecipe.getName().equals(getName())
                 && otherRecipe.getInstruction().equals(getInstruction())
                 && otherRecipe.getRecipeImage().equals(getRecipeImage())
-                && otherRecipe.getIngredient().equals(getIngredient())
+                && isSameIngredients(otherRecipe.getIngredient())
                 && otherRecipe.getCalories().equals(getCalories())
                 && otherRecipe.getTags().equals(getTags());
     }
 
+    private boolean isSameIngredients(ArrayList<Ingredient> otherIngredients) {
```
Comment (caitlinjee) [inline src/main/java/seedu/address/model/recipe/Recipe.java:102]: does this check for different ordering of ingredients?

File: src/main/java/seedu/address/model/recipe/Tag.java [hunk 3]
```diff
@@ -21,6 +21,7 @@
      */
     public Tag(String tagName) {
         requireNonNull(tagName);
+        System.out.println(isValidTagName(tagName));
```
Comment (caitlinjee) [inline src/main/java/seedu/address/model/recipe/Tag.java]: supposed to remove this line?

File: src/main/java/seedu/address/logic/commands/recipe/AddRecipeCommand.java (hunk 1 above)
Comment (oliviajohansen) [reply src/main/java/seedu/address/logic/commands/recipe/AddRecipeCommand.java]: ok

File: src/main/java/seedu/address/model/recipe/Tag.java (hunk 3 above)
Comment (oliviajohansen) [reply src/main/java/seedu/address/model/recipe/Tag.java]: yes just realised

File: src/main/java/seedu/address/model/recipe/Recipe.java (hunk 2 above)
Comment (oliviajohansen) [reply src/main/java/seedu/address/model/recipe/Recipe.java:102]: no should I check for diff ordering?

**Review by daongochieu2810 — APPROVED** [review]
```

```

**Review by caitlinjee — APPROVED** [review]
lgtm

**Review by TianYong-Goh — APPROVED** [review]
lgtm

**Review by jiaqi20 — APPROVED** [review]
lgtm

**Pull request merged**